plugins/modules/ec2_instance.py

```diff
@@ -1389,28 +1392,37 @@ def find_instances(ec2, ids=None, filters=None):
                 filters[key.replace("_", "-")] = filters.pop(key)
         params = dict(Filters=ansible_dict_to_boto3_filter_list(filters))
 
-    results = paginator.paginate(**params).search('Reservations[].Instances[]')
+    try:
+        results = paginator.paginate(**params).search('Reservations[].Instances[]')
+    except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError) as e:
+        module.fail_json_aws(e, msg="Could not describe instances")
```

Comment:
Catching the Error here prevents it falling through the the jittered backoff and being retried it's probably worth  rejigging this a little:

```suggestion
    try:
        results = _describe_instances(ec2, **params)
    except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError) as e:
        module.fail_json_aws(e, msg="Could not describe instances")


@AWSRetry.jittered_backoff()
def _describe_instances(ec2, **params):   
    paginator = ec2.get_paginator('describe_instances')
    paginator.paginate(**params).search('Reservations[].Instances[]')
```